tests/test_arthur.py

```diff
@@ -62,49 +63,208 @@ def setUp(self):
+        self.assertEqual(t.archive_args['fetch_from_archive'], archive_params["fetch_from_archive"])
+        self.assertEqual(t.archive_args['archived_after'], archive_params["archived_after"])
+        self.assertEqual(t.archive_args['archive_path'], ARCHIVES_DEFAULT_PATH)
+
+    def test_task_wrong_fetch_from_archive(self):
+        """Check whether an exception is thrown when fetch_from_archive parameter is not properly set"""
+
+        task_id = "arthur.task"
+        backend = "backend"
+        category = None
+        backend_params = {"a": "a", "b": "b"}
+        archive_params = {"fetch_from_archive": 100, "archived_after": "2010-10-10"}
+
+        app = Arthur(self.conn, async_mode=False)
+
+        with self.assertRaises(ValueError):
+            app.add_task(task_id, backend, category, backend_params, archive_params)
+
+        archive_params = {"archived_after": "2010-10-10"}
+        with self.assertRaises(ValueError):
+            app.add_task(task_id, backend, category, backend_params, archive_params)
+
+    def test_task_ignore_archive_after(self):
+        """Check whether the archived_after parameter is not set when fetch_from_archive is false"""
+
+        task_id = "arthur.task"
+        backend = "backend"
+        category = None
+        backend_params = {"a": "a", "b": "b"}
+        archive_params = {"fetch_from_archive": False, "archived_after": "X"}
+
+        app = Arthur(self.conn, async_mode=False)
+        app.add_task(task_id, backend, category, backend_params, archive_params)
+
+        t = app._tasks.tasks[0]
+        self.assertEqual(t.archive_args['fetch_from_archive'], False)
+        self.assertIsNone(t.archive_args['archived_after'])
+
+    def test_task_wrong_archive_after(self):
+        """Check whether an exception is thrown when archived_after parameter is not properly set"""
+
+        task_id = "arthur.task"
+        backend = "backend"
+        category = None
+        backend_params = {"a": "a", "b": "b"}
+        archive_params = {"fetch_from_archive": True, "archived_after": "X"}
+
+        app = Arthur(self.conn, async_mode=False)
+
+        with self.assertRaises(ValueError):
+            app.add_task(task_id, backend, category, backend_params, archive_params)
+
+        archive_params = {"fetch_from_archive": True}
+        with self.assertRaises(ValueError):
+            app.add_task(task_id, backend, category, backend_params, archive_params)
+
+    def test_add_task_archive(self):
+        """Check whether tasks are added"""
+
+        task_id = "arthur.task"
+        backend = "backend"
+        category = 'acme-product'
+        backend_params = {"a": "a", "b": "b"}
+        archive_params = {'fetch_from_archive': True,
+                          'archived_after': '2100-01-01',
+                          'archive_path': './acme-plan'}
+        sched_params = {"delay": 10, "max_retries_job": 10}
+
+        app = Arthur(self.conn, base_archive_path='./acme-plan', async_mode=False)
```

Comment:
I'd rather use a temporal directory created by python standard library. Check the archive tests in Perceval.